src/Shouldly/ShouldlyExtensionMethods/ShouldBeBooleanExtensions.cs

```diff
@@ -8,33 +9,33 @@ namespace Shouldly
     [ShouldlyMethods]
     public static class ShouldBeBooleanExtensions
     {
-        public static void ShouldBeTrue(this bool actual)
+        public static void ShouldBeTrue([DoesNotReturnIf(false)] this bool actual)
```

Comment:
`DoesNotReturnIf` is the annotation that powers scenario like this:

```cs
nullableThing.GetHashCode(); // Compiler warning, possible NRE
(nullableThing != null).ShouldBeTrue();
nullableThing.GetHashCode(); // No warning because ShouldBeTrue did not throw an exception
```